time_tracker_api/projects/projects_model.py

```diff
@@ -1,80 +1,86 @@
-from time_tracker_api.errors \
-    import MissingResource, InvalidInput, InvalidMatch
+import enum
 
+from flask import Flask
 
-class InMemoryProjectDAO(object):
-    def __init__(self):
-        self.counter = 0
-        self.projects = []
+from time_tracker_api.database import DATABASE_TYPE, CRUDDao
+from time_tracker_api.sql_repository import SQLCRUDDao, SQLAuditedModel, SQLModel
 
-    def get_all(self):
-        return self.projects
 
-    def get(self, id):
-        for project in self.projects:
-            if project.get('id') == id:
+class PROJECT_TYPE(enum.Enum):
+    CUSTOMER = 'CUSTOMER'
+    TRAINING = 'TRAINING'
+
+    @classmethod
+    def valid_type_values(self):
+        return list(map(lambda x: x.value, PROJECT_TYPE._member_map_.values()))
+
+
+class ProjectDao(CRUDDao):
+    pass
+
+
+def create_dao(app: Flask) -> ProjectDao:
+    """This will construct the dao based on the chosen DATABASE"""
+    if app.config['DATABASE'] == DATABASE_TYPE.SQL:
+        from time_tracker_api.sql_repository import db
+
+        class ProjectSQLModel(db.Model, SQLModel, SQLAuditedModel):
+            __tablename__ = 'projects'
+            id = db.Column(db.Integer, primary_key=True)
+            name = db.Column(db.String(50), unique=True, nullable=False)
+            description = db.Column(db.String(250), unique=False, nullable=False)
+            type = db.Column(db.String(10), nullable=False)
+            active = db.Column(db.Boolean, default=True)
+
+            def __repr__(self):
+                return '<Project %r>' % self.name
+
+            def __str___(self):
+                return "the project \"%s\"" % self.name
+
+        class ProjectSQLDao(SQLCRUDDao):
+            def __init__(self):
+                SQLCRUDDao.__init__(self, ProjectSQLModel)
+
+        return ProjectSQLDao()
+    else:
+        """
```

Comment:
I think we should get rid of all the in-memory implementation. 
1. We will need to implement this in-memory code for all DAOs
2. We are not going to need this in-memory solution
3. A Dao should has more implementations if we really need them. For now, just one sql implementation is good enough

Reply:
This in-memory code will not be used, this is just basically for documentation purposes. I wanted the other people who work in the project to see how you can create other db mechanisms. And now that the Azure database serverless implementation is showing some issues, it is nice to be aware of how this work.